src/Compilers/CSharp/Portable/BoundTree/BoundNode.cs

```diff
@@ -170,29 +170,19 @@ public void ResetCompilerGenerated(bool newCompilerGenerated)
         /// <summary>
         /// Top level nullability for the node. This should not be used by flow analysis.
         /// </summary>
+        [DebuggerHidden]
         protected NullabilityInfo TopLevelNullability
```

Comment:
>NullabilityInfo [](start = 18, length = 15)

Consider exposing the internal `NullableAnnotation` and `NullableFlowState` values and have the `SemanticModel` be the one place where internal values are mapped to the public enum values.

Reply:
I prefer to do it this way.

---
In reply to: [269156210](https://github.com/dotnet/roslyn/pull/34429#discussion_r269156210) [](ancestors = 269156210)

Reply:
Ideally it should be possible to implement the analysis without converting from the public enum to the internal enum.

---
In reply to: [269762381](https://github.com/dotnet/roslyn/pull/34429#discussion_r269762381) [](ancestors = 269762381,269156210)